<launch>
  <arg name="map_frame" default="map"/>

  <node pkg="project11_navigation" type="navigator" name="navigator">
    <remap from="~/cmd_vel" to="project11/piloting_mode/autonomous/cmd_vel"/>
    <remap from="~/enable" to="project11/piloting_mode/autonomous/active"/>
  </node>
  <node pkg="project11_navigation" type="occupancy_vector_map_from_geo" name="occupancy_vector_map_from_geo">
    <remap from="~/input" to="project11/avoidance_map"/>
    <remap from="~/output" to="project11/avoidance_map_local"/>
    <param name="frame_id" value="$(arg map_frame)"/>
  </node>
  <node pkg="project11_navigation" type="occupancy_grid_from_vector_map" name="occupancy_grid_from_vector_map">
    <remap from="~/input" to="project11/avoidance_map_local"/>
    <remap from="~/output" to="project11/avoidance_grid"/>
  </node>

  <node pkg="manda_coverage" type="manda_coverage_node" name="manda_coverage">
    <param name="soundings_topic" value="sensors/mbes/soundings"/>
  </node>

  <rosparam file="$(find project11_navigation)/config/default_stack.yaml" command="load"/>

</launch>
